api/generator.py

```diff
@@ -0,0 +1,77 @@
+import face_recognition
+import os
+import cv2
+
+
+KNOWN_FACES_DIR = 'api/known_faces'
+TOLERANCE = 0.6
+FRAME_THICKNESS = 3
+FONT_THICKNESS = 1
+MODEL = 'cnn'
+
+known_faces = []
+known_names = []
+
+for name in os.listdir(KNOWN_FACES_DIR):
+    try:
+        for filename in os.listdir(f'{KNOWN_FACES_DIR}/{name}'):
+            image = face_recognition.load_image_file(f"{KNOWN_FACES_DIR}/{name}/{filename}")
+            encoding = face_recognition.face_encodings(image)[0]
+            known_faces.append(encoding)
+            known_names.append(name)
+    except NotADirectoryError:
+        pass
+
+
+def gen():
+    # when you use a web camera
+    video = cv2.VideoCapture(0)
+
+    # when you want to use video file
+    # video = cv2.VideoCapture('api/static/IMG_0909.MOV')
+
+    # Read until video is completed
+    while(video.isOpened()):
+        # Find OpenCV version
+        (major_ver, minor_ver, subminor_ver) = (cv2.__version__).split('.')
+        if int(major_ver) < 3:
+            fps = video.get(cv2.cv.CV_CAP_PROP_FPS)
+        else:
+            fps = video.get(cv2.CAP_PROP_FPS)
+        num_frames = 60
```

Comment:
I suppose that FPS should be available somewhere in video-file metadata. Would you create an issue for this?

Reply:
Yes, I will create an issue.
Thank you for the comments.
We'll keep in touch.